gc/base/GCExtensionsBase.cpp

```diff
@@ -182,10 +182,12 @@ MM_GCExtensionsBase::initialize(MM_EnvironmentBase* env)
 	if (J9HookInitializeInterface(getPrivateHookInterface(), OMRPORTLIB, sizeof(privateHookInterface))) {
 		goto failed;
 	}
+	INITIALIZE_HOOKDUMP(privateHookInterface);
```

Comment:
typical HookInterface like below
typedef struct xxxxHookInterface {
	struct J9CommonHookInterface common;
	U_8 flags[xx];
	struct EventInfo4Dump infos4Dump[xx];
	J9HookRecord* hooks[xx];
} xxxxHookInterface;

to access flags and hooks, we use the below macros, accessing flags from the end of "common" interface, and accessing hooks from end of the interface in descending order, without knowing array size, we can not locate start address of EventInfo4Dump array.  create new macro J9HOOK_INITIALIZE_INTERFACE_AND_DUMP instead of J9HookInitializeInterface() to avoid missing the extra macro step. 

/* flags are stored at the beginning of the interface just after the common interface fields in ascending order */
#define HOOK_FLAGS(interface, event) (((uint8_t*)((interface) + 1))[event])
/* records are stored at the END of the interface in descending order */
#define HOOK_RECORD(interface, event) (((J9HookRecord**)( (uint8_t*)(interface) + (interface)->size ))[ -1 - (event)])